.github/workflows/sync-backport-changelog.yml

```diff
@@ -0,0 +1,70 @@
+name: Sync Core Backport Issue
+
+on:
+    push:
+        branches:
+            - trunk
+
+jobs:
+    sync-backport-changelog:
+        name: Sync Core Backport Issue
+        runs-on: ubuntu-latest
+        steps:
+            - uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
+            - name: Sync Issue
+              uses: actions/github-script@60a0d83039c74a4aee543508d2ffcb1c3799cdea # v7.0.1
+              with:
+                  script: |
+                      const labelName = '🤖 Sync Backport Changelog';
+                      const issues = await github.paginate(github.rest.issues.listForRepo, {
+                        owner: context.repo.owner,
+                        repo: context.repo.repo,
+                        labels: [labelName],
+                        state: 'open',
+                        per_page: 1,
+                      });
+
+                      if (issues.length === 0) {
+                        console.log(`No issues found with the "${labelName}" label.`);
+                        return;
+                      }
+
+                      const [latestIssue] = issues;
+                      const versionMatch = latestIssue.title.match(/(\d+\.\d+)/);
+                      if (!versionMatch) {
+                        console.log('Could not find a version number in the latest issue title.');
+                        return;
+                      }
+
+                      const version = versionMatch[1];
+                      console.log(`Latest version: ${version}`);
+
+                      const { execSync } = require('child_process');
+                      const processedChangelog = execSync(`awk '/./ {print ($0 ~ /^[-*]/ ? "  " : "- ") $0}' backport-changelog/${version}/*.md`).toString().trim();
```

Comment:
Could we check the last commit for changes in this directory before doing anything else? Seems like it might save us some API requests.
Another alternative could be conditioning this workflow on a check similar to the one `check-backport-changelog` uses.

Reply:
What API requests would it save? I'm already checking if the content is different.

Reply:
Added an early abort 8bb22a2b43fc439fcdb67b2151ffb80fb4a3ba02